engines/sherlock/animation.cpp

```diff
@@ -0,0 +1,203 @@
+	const char *p = strchr(baseName.c_str(), '.');
+	if (p)
+		baseName = Common::String(filename.c_str(), MIN(p - 1, baseName.c_str() + 7));
+
+	Common::String vdxName = baseName + ".vdx";
+
+	// Load the animation
+	Common::SeekableReadStream *stream;
+	if (!_vm->_titleOverride.empty())
+		stream = _vm->_res->load(vdxName, _vm->_titleOverride);
+	else if (_vm->_useEpilogue2)
+		stream = _vm->_res->load(vdxName, "epilog2.lib");
+	else
+		stream = _vm->_res->load(vdxName, "epilogue.lib");
+
+	// Load initial image
+	Common::String vdaName = baseName + ".vda";
+	ImageFile images(vdaName, true, true);
+
+	events.wait(minDelay);
+	if (fade != 0 && fade != 255)
+		screen.fadeToBlack();
+
+	if (setPalette) {
+		if (fade != 255)
+			screen.setPalette(images._palette);
+	}
+
+	int frameNumber = 0;
+	Common::Point pt;
+	bool skipped = false;
+	while (!_vm->shouldQuit()) {
+		// Get the next sprite to display
+		int imageFrame = stream->readSint16LE();
+
+		if (imageFrame == -2) {
+			// End of animation reached
+			break;
+		} else if (imageFrame != -1) {
+			// Read position from either animation stream or the sprite frame itself
+			if (imageFrame < 0) {
+				imageFrame += 32768;
+				pt.x = stream->readUint16LE();
+				pt.y = stream->readUint16LE();
+			} else {
+				pt = images[imageFrame]._offset;
+			}
+
+			// Draw the sprite. Note that we explicitly use the raw frame below, rather than the ImageFrame,
+			// since we don't want the offsets in the image file to be used, just the explicit position we specify
+			screen.transBlitFrom(images[imageFrame]._frame, pt);
+		} else {
+			// No sprite to show for this animation frame
+			if (fade == 255) {
+				// Gradual fade in
+				if (screen.equalizePalette(images._palette) == 0)
+					fade = 0;
+			}
+
+			// Check if we've reached a frame with sound
+			if (frameNumber++ == *soundFrames) {
+				++soundNumber;
+				++soundFrames;
+				Common::String fname = _vm->_soundOverride.empty() ?
+					Common::String::format("%s%01d", baseName.c_str(), soundNumber) :
+					Common::String::format("%s%02d", baseName.c_str(), soundNumber);
+
+				if (sound._voices)
+					sound.playSound(fname);
+			}
+
+			events.wait(speed);
+		}
+
+		if (events.kbHit()) {
+			Common::KeyState keyState = events.getKey();
+			if (keyState.keycode == Common::KEYCODE_ESCAPE ||
+				keyState.keycode == Common::KEYCODE_SPACE) {
+				skipped = true;
+				break;
+			}
+		} else if (events._pressed) {
+			skipped = true;
+			break;
+		}
+	}
+
+	events.clearEvents();
+	sound.stopSound();
+	delete stream;
+	sound._playingEpilogue = false;
+
+	return !skipped && !_vm->shouldQuit();
+}
+
+/**
+ * Checks for whether an animation is being played that has associated sound
+ */
+const int *Animation::checkForSoundFrames(const Common::String &filename) {
+	const int *frames = &NO_FRAMES;
+
+	if (_vm->_soundOverride.empty()) {
+		for (int idx = 0; idx < PROLOGUE_NAMES_COUNT; ++idx) {
+			if (!scumm_stricmp(filename.c_str(), PROLOGUE_NAMES[idx])) {
```

Comment:
Using filename.equalsIgnoreCase(PROLOGUE_NAMES[idx]) may be cleaner (and less C-ish).